commands/src/main/java/org/wildfly/extras/creaper/commands/elytron/CreateServerSSLContext.java

```diff
@@ -0,0 +1,386 @@
+package org.wildfly.extras.creaper.commands.elytron;
+
+import java.util.Arrays;
+import java.util.List;
+
+import javax.net.ssl.KeyManagerFactory;
+
+import org.wildfly.extras.creaper.commands.elytron.CredentialRef.CredentialRefBuilder;
+import org.wildfly.extras.creaper.commands.elytron.tls.AddKeyManager;
+import org.wildfly.extras.creaper.commands.elytron.tls.AddKeyStore;
+import org.wildfly.extras.creaper.commands.elytron.tls.AddServerSSLContext;
+import org.wildfly.extras.creaper.commands.elytron.tls.AddTrustManager;
+import org.wildfly.extras.creaper.core.online.OnlineCommand;
+import org.wildfly.extras.creaper.core.online.OnlineCommandContext;
+
+/**
+ * Composite creaper command. Takes flat parameters and compose server ssl
+ * context with all dependent resource hierarchy.
+ *
+ * <p>
+ * It should be possible to configure most of use cases atomic creaper commands
+ * allows. But ther are some limitation as provider and providerLoader if set
+ * are used globally for all resources which can specify that (key-store,
+ * key-manager, trust-manager)
+ * </p>
+ *
+ */
+public class CreateServerSSLContext implements OnlineCommand {
+
+    private static final String KEY_STORE_NAME = "key-store-name";
+    private static final String TRUST_STORE_NAME = "trust-store-name";
+    private static final String KEY_MANAGER_NAME = "key-manager-name";
+    private static final String TRUST_MANAGER_NAME = "trust-manager-name";
+
+    // SSL Context
+    private final String name;
+    protected final List<String> protocols;
+    private final String cipherSuiteFilter;
+    private final Boolean needClientAuth;
+    private final Boolean wantClientAuth;
+    private final Boolean authenticationOptional;
+    private final String securityDomain;
+    private final Integer maximumSessionCacheSize;
+    private final Integer sessionTimeout;
+    // Keystore
+    private final String keyStoreType;
+    private final String keyStorePath;
+    private final String keyStorePassword;
+    private final String keyPassword;
+    private final String keyStoreAlias;
+    private final String keyStoreRelativeTo;
+    private final Boolean keyStoreRequired;
+    // Truststore
+    private final String trustStoreType;
+    private final String trustStorePath;
+    private final String trustStorePassword;
+    private final String trustStoreAlias;
+    private final String trustStoreRelativeTo;
+    private final Boolean trustStoreRequired;
+    // Multiple usage
+    private final String algorithm;         // keystore manager, truststore manager
+    private final String provider;          // keystore manager, truststore manager, keystore
```

Comment:
In which cases do we need common custom provider for keystore manager, truststore manager, and keystore?